timeseries/src/autogluon/timeseries/predictor.py

```diff
@@ -194,7 +177,9 @@ def __init__(
         self.freq = freq
         if self.freq is not None:
             # Standardize frequency string (e.g., "T" -> "min", "Y" -> "YE")
-            std_freq = pd.tseries.frequencies.to_offset(self.freq).freqstr
+            offset = pd.tseries.frequencies.to_offset(self.freq)
+            assert offset is not None
+            std_freq = offset.freqstr
```

Comment:
Disappointing that `pyright` cannot infer that `offset is not None` from the method signature :(

![image](https://github.com/user-attachments/assets/f279fb50-63e1-44f3-803e-b633d7867b02)